### DISCUSSION TITLE
Fixing OBS support on MacOS 14 - Need help

### DISCUSSION THREAD

**Discussion by GermanWarez** [letmaik/pyvirtualcam#119] [General]
I'd like to fix the bug https://github.com/letmaik/pyvirtualcam/issues/111

How would I do this? Should I create a fork (similar to the AkVirtualCam)?
Disclaimer: I did wrote a single line of C++ code, so I'll rely on ChatGPT and Github Copilot (which is a pain, since these language models just reproduce code that looks relevant in the context of my request). It'll be hard, but with the infinite wisdom (gg) of these helpers it seems feasible.

**Comment by letmaik** [comment]
I'm not sure what your reference to AkVirtualCam means, but yes, create a fork and do your changes in a branch. If it's working locally and CI passes, submit a PR.

**Comment by GermanWarez** [comment]
Technically not possible, due to security restrictions for service extensions.
For more details see the bug https://github.com/letmaik/pyvirtualcam/issues/111#issuecomment-1979304623

My experience with CoPilot: It's quite good to explain functions, but itself it can't really fix anything. Eventually I got all code working, except for a segfault that would always happen when opening the camera in python:  In the source code it's at the point where the string with the UUID is compared. And then I realized I cannot access the virtual camera. But, learned quite a bit about C/ObJC and I remembered why I use scripts instead of compiled languages most of the time.
